phpunit/class-wp-theme-json-resolver-test.php

```diff
@@ -226,6 +226,83 @@ public function test_add_theme_supports_are_loaded_for_themes_without_theme_json
 		$this->assertSame( $color_palette, $settings['color']['palette']['theme'] );
 	}
 
+	public function test_add_registered_block_styles_to_theme_data() {
+		switch_theme( 'block-theme' );
+
+		$variation_styles_data = array(
+			'color'    => array(
+				'background' => 'darkslateblue',
+				'text'       => 'lavender',
+			),
+			'blocks'   => array(
+				'core/heading' => array(
+					'color' => array(
+						'text' => 'violet',
+					),
+				),
+			),
+			'elements' => array(
+				'link' => array(
+					'color'  => array(
+						'text' => 'fuchsia',
+					),
+					':hover' => array(
+						'color' => array(
+							'text' => 'deeppink',
+						),
+					),
+				),
+			),
+		);
+
+		register_block_style(
+			'core/group',
+			array(
+				'name'       => 'my-variation',
+				'style_data' => $variation_styles_data,
+			)
+		);
+
+		$theme_json   = WP_Theme_JSON_Resolver_Gutenberg::get_theme_data()->get_raw_data();
+		$group_styles = $theme_json['styles']['blocks']['core/group'] ?? array();
+		$expected     = array(
+			'variations' => array(
+				'my-variation' => $variation_styles_data,
+			),
+		);
+
+		$this->assertSameSetsWithIndex( $group_styles, $expected );
+
+		unregister_block_style( 'core/group', 'my-variation' );
```

Comment:
Tiny nit: if the assertion fails, does `unregister_block_style` still get called? If not, would it be worth moving the unregister calls to just prior to the assertions in these tests?

Reply:
Thanks. ✅ (https://github.com/WordPress/gutenberg/pull/56540/commits/611913f8a06b073e961cc1df4c5f44879c40e44a)